src/utils/config.rs

```diff
@@ -0,0 +1,36 @@
+use core::cmp;
+
+//------------ DefMinMax -----------------------------------------------------
+
+/// The default, minimum, and maximum values for a config variable.
+#[derive(Clone, Copy)]
+pub struct DefMinMax<T> {
+    /// The default value,
+    def: T,
+
+    /// The minimum value,
+    min: T,
+
+    /// The maximum value,
+    max: T,
+}
+
+impl<T> DefMinMax<T> {
+    /// Creates a new value.
+    pub const fn new(def: T, min: T, max: T) -> Self {
```

Comment:
I have a slignt preference for min, def, max instead of def, min, max because I think of def being a value in between min and max.

Reply:
Ah, my bad.